frontend/packages/component-library/src/actionBlock/stories/ActionBlock.story.tsx

```diff
@@ -0,0 +1,321 @@
+import image3 from '@public/images/action-block-03.png';
+import type {Meta, StoryObj} from '@storybook/react';
+import {within, expect} from '@storybook/test';
+
+import ActionBlock, {FullWidthActionBlock, ActionBlockProps} from '../index';
+
+export default {
+  title: 'DesignSystem/Action Block',
+  component: ActionBlock,
+  parameters: {
+    a11y: {
+      config: {
+        rules: [
+          {
+            // Disable the color contrast rule for action blocks.
+            // ActionBlock component has one a11y issue, and it's related to the overline color.
+            // This is a known issue across our design system, and we are ok accepting this for now.
+            id: 'color-contrast',
+            enabled: false,
+          },
+        ],
+      },
+    },
+  },
+} as Meta;
+type Story = StoryObj<typeof ActionBlock>;
+
+const DESCRIPTION =
+  'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eget risus vitae massa semper aliquam quis mattis quam.';
+const DESCRIPTION_MED =
+  'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eget risus vitae massa semper aliquam quis mattis quam. Lorem ipsum dolor sit amet, consectetur adipiscing elit.';
+const DESCRIPTION_LONG =
+  'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eget risus vitae massa semper aliquam quis mattis quam. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eget risus vitae massa semper aliquam quis mattis quam.';
+
+const defaultArgs: ActionBlockProps = {
+  title: 'Action block title',
+  description: DESCRIPTION,
+  image: image1,
+  overline: 'Overline Text',
+  background: 'primary',
+  primaryButton: {
+    text: 'Primary Button',
+    href: '#',
+    ariaLabel: 'Primary Button aria label',
+  },
+  secondaryButton: {
+    text: 'Secondary Button',
+    href: '#',
+    ariaLabel: 'Secondary Button aria label',
+  },
+};
+
+//
+// STORIES
+//
+export const DefaultActionBlocks: Story = {
+  args: {
+    ...defaultArgs,
+  },
+  render: args => {
+    return (
+      <div
+        style={{
+          display: 'grid',
+          gridTemplateColumns: 'repeat(2, 1fr)',
+          gap: '1.5rem',
+        }}
+      >
+        <ActionBlock {...args} />
+        <ActionBlock {...args} image={image2} />
+      </div>
+    );
+  },
+  play: async ({canvasElement}) => {
+    const canvas = within(canvasElement);
+    const titles = await canvas.findAllByText('Action block title');
+    const descriptions = await canvas.findAllByText(DESCRIPTION);
+    const overlines = await canvas.findAllByText('Overline Text');
+    const images = await canvas.findAllByRole('figure');
+    const primaryButtons = await canvas.findAllByLabelText(
+      'Primary Button aria label',
+    );
+    const secondaryButtons = await canvas.findAllByLabelText(
+      'Secondary Button aria label',
+    );
+
+    // check if images are visible
+    await expect(images).toHaveLength(2);
```

Comment:
Would be nice to find these by alt text

Reply:
See [comment](https://github.com/code-dot-org/code-dot-org/pull/64639#discussion_r2008377572) above re: decorative images.

Reply:
Although I just updated the unit tests to use empty alt text, I can update that here too.